semantic-analysis/src/Analysis/Analysis/Exception.hs

```diff
@@ -47,6 +62,35 @@ exc :: Exception -> ExcSet
 exc e = ExcSet mempty (Set.singleton e)
 
 
+exceptionTracing
+  :: Ord term
+  => ( forall sig m
+     .  (Has (Env A.MAddr) sig m, HasLabelled Store (Store A.MAddr ExcSet) sig m, Has (Dom ExcSet) sig m, Has A.Statement sig m)
+     => (term -> m ExcSet)
+     -> (term -> m ExcSet) )
+  -> [File term]
+  -> (A.MStore ExcSet, [File (Module ExcSet)])
+exceptionTracing eval = A.runFiles (runFile eval)
+
+runFile
+  :: ( Has (State (A.MStore ExcSet)) sig m
+     , Ord term )
+  => ( forall sig m
+     .  (Has (Env A.MAddr) sig m, HasLabelled Store (Store A.MAddr ExcSet) sig m, Has (Dom ExcSet) sig m, Has A.Statement sig m)
+     => (term -> m ExcSet)
+     -> (term -> m ExcSet) )
+  -> File term
+  -> m (File (Module ExcSet))
```

Comment:
Note that this is the first of our analyses to return a `Module` (and thus to support modular analysis). Unfortunately, because we don't yet track exports, we can't do linking properly.